org/qmckl_sherman_morrison_woodbury.org

```diff
@@ -85,33 +86,33 @@ int main() {
 
     #+RESULTS:
     #+begin_src c :tangle (eval h_func) :comments org
-    qmckl_exit_code qmckl_sherman_morrison_c (
+    qmckl_exit_code qmckl_sherman_morrison(
 	  const qmckl_context context,
-	  const uint64_t Dim,
-	  const uint64_t N_updates,
+	  const uint64_t* Dim_p,
+	  const uint64_t* N_updates_p,
 	  const double* Updates,
 	  const uint64_t* Updates_index,
-    const double breakdown,
-	  double* Slater_inv ); 
+	  const double* breakdown_p,
+	  double* Slater_inv); 
     #+end_src
 
-*** Source
+*** C source
 
     #+begin_src c :tangle (eval c) :comments org
 #include <stdbool.h>
 #include <math.h>
 #include "qmckl.h"
 
-qmckl_exit_code qmckl_sherman_morrison_c(const qmckl_context context, 
-                                const uint64_t Dim,
-                                const uint64_t N_updates,
+qmckl_exit_code qmckl_sherman_morrison_c_(const qmckl_context context, 
+                                const uint64_t* Dim_p,
+                                const uint64_t* N_updates_p,
                                 const double* Updates,
                                 const uint64_t* Updates_index,
-                                const double breakdown,
-                                double * Slater_inv) {
-// #ifdef DEBUG
-//   std::cerr << "Called qmckl_sherman_morrison with " << N_updates << " updates" << std::endl;
-// #endif
+                                const double* breakdown_p,
+                                double* Slater_inv) {
+  const uint64_t Dim = *Dim_p;
+  const uint64_t N_updates = *N_updates_p;
+  const double breakdown = *breakdown_p;
 
```

Comment:
@fmgjcoppens One thing that would be nice would be to add asserts to make sure that these pointers are not NULL or 0 just in case. This way it is easy to debug later. Just a suggestion.

Reply:
You mean in the tests, or in the kernels themselves. Won't this slow down the kernel?